### DISCUSSION TITLE
How can I handling response status globally?

### DISCUSSION THREAD

**Discussion by positivedeveloper36** [adonisjs/core#3831] [Help]
How can I handling response status globally?

My controller like this :
```
  public async store({ request, response }: HttpContextContract) {
    const input = request.only(['username', 'email'])
    //try {
      const users = await User.create(input)

    //  return response.status(200).json({ code: 200, status: 'success', data: users })
    //} catch (err) {
    //  return response.status(404).json({ code: 404, status: 'error', message: err.message })
    //}
  };
```

I handling exceptions globally using exception handler like this :

```
export default class ExceptionHandler extends HttpExceptionHandler {
  constructor() {
    super(Logger)
  }

  public async handle(error: any, ctx: HttpContextContract) {
    if (error.code === 'E_ROUTE_NOT_FOUND') {
      return ctx.response.notFound({
        error: [{ message: 'Resource not found'}]
      })
    }
    else if (error.code === 'E_VALIDATION_FAILURE') {
      return ctx.response.status(422).send(error.messages)
    }

    return super.handle(error, ctx)
  }
}
```

It is located in app\Exception\Handler.ts

The problem is that it only handles errors globally, it can't handle success response globally
How can I make a function so that it also can handle success response globally?

Please help. Thank you very much

**Comment by Barbapapazes** [comment]
Hello, 

Can you provide more information about your use case? I can't see how your fonction are related to a 'E_ROW_NOT_FOUND', can you explain?

Thanks

**Reply by positivedeveloper36** [reply]
@Barbapapazes I had updated my question

**Reply by Barbapapazes** [reply]
Yeah thanks! What do you call 'success'? Is it to return data to client?

**Reply by positivedeveloper36** [reply]
Yeah like this ```return response.status(200).json({ code: 200, status: 'success', data: users })``` So I want to make a function that can handle the success response and error response. So it can be used in many controllers

**Reply by Barbapapazes** [reply]
juste return users! Adonis handle this for you!

**Reply by Barbapapazes** [reply]
the statusCode will be in http so you don't have to add it to your response (and status will be 'ok')

**Reply by positivedeveloper36** [reply]
Ok thank you very much
